ConsistencyManagerTests/ModelUpdatesListenerTests.swift

```diff
@@ -0,0 +1,116 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import XCTest
+@testable import ConsistencyManager
+
+class ModelUpdatesListenerTests: ConsistencyManagerTestCase {
+    
+    func testSingleListener() {
+        let requiredModel = TestRequiredModel(id: "1", data: 0)
+        let testModel = TestModel(id: "0", data: 0, children: [], requiredModel: requiredModel)
+
+        let consistencyManager = ConsistencyManager()
+        let listener = TestUpdatesListener()
+
+        addUpdatesListener(listener, toConsistencyManager: consistencyManager)
+
+        var calledUpdateClosure = 0
+        listener.updateClosure = { model, children, context in
+            calledUpdateClosure += 1
+            XCTAssertEqual(model as? TestModel, testModel)
+            XCTAssertEqual(children.count, 2)
+            XCTAssertEqual(children["0"]??[0] as? TestModel, testModel)
+            XCTAssertEqual(children["1"]??[0] as? TestRequiredModel, requiredModel)
+            XCTAssertEqual(context as? String, "context")
+        }
+
+        updateNewModel(testModel, consistencyManager: consistencyManager, context: "context")
+
+        XCTAssertEqual(calledUpdateClosure, 1)
+    }
+
+    func testMultipleListeners() {
+        let requiredModel = TestRequiredModel(id: "1", data: 0)
+        let testModel = TestModel(id: "0", data: 0, children: [], requiredModel: requiredModel)
+
+        let consistencyManager = ConsistencyManager()
+        let listener = TestUpdatesListener()
+        let listener2 = TestUpdatesListener()
+
+        addUpdatesListener(listener, toConsistencyManager: consistencyManager)
+        addUpdatesListener(listener2, toConsistencyManager: consistencyManager)
+
+        var calledUpdateClosure = 0
+        listener.updateClosure = { model, children, context in
+            calledUpdateClosure += 1
+            XCTAssertEqual(model as? TestModel, testModel)
+            XCTAssertEqual(children.count, 2)
+            XCTAssertEqual(children["0"]??[0] as? TestModel, testModel)
+            XCTAssertEqual(children["1"]??[0] as? TestRequiredModel, requiredModel)
```

Comment:
Lol, actually, it doesn't work with spaces. This isn't the `??` operator. It's two `?``?` operators because it's a double optional.

Making this cleaner in a future commit, but with this API, this is actually the only correct way.